src/apps/settings/src/app/views/User/Workflows/authorized/forms-dialogs/DeactivationDialog.tsx

```diff
@@ -0,0 +1,118 @@
+import { FC, useState, useCallback } from "react";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import PauseCircleOutlineRoundedIcon from "@mui/icons-material/PauseCircleOutlineRounded";
+import Typography from "@mui/material/Typography";
+import { useDispatch } from "react-redux";
+import { notify } from "../../../../../../../../../shell/store/notifications";
+import { LoadingButton } from "@mui/lab";
+import { useDeactivateWorkflowStatusLabelMutation } from "../../../../../../../../../shell/services/instance";
+
+type DeactivationDialogProps = {
+  open: boolean;
+  onClose: () => void;
+  name: string;
+  ZUID: string;
+  callBack?: () => void;
+};
+
+const DeactivationDialog: FC<DeactivationDialogProps> = ({
+  open,
+  onClose,
+  name,
+  ZUID,
+  callBack,
+}) => {
+  const dispatch = useDispatch();
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
this one is actually intended for the button's loading indicator.

Reply:
The query's "isLoading" variable is out of scope. It is triggered inside a popup modal, where it controls the loading state for the asynchronous "deactivateWorkflowStatusLabel" function. This ensures that the modal does not close and displays a loading indicator on the button while the function is being executed.

Reply:
Oh, I didn't realize that. I actually tried to look into it because I thought the same thing but couldn't find any references (or maybe I didn't search thoroughly enough). Thanks, Andres!